src/components/dialogs/create-study-dialog/create-study-dialog.js

```diff
@@ -0,0 +1,388 @@
+                            createStudyFormMethods.getValues().caseUuid;
+
+                        if (prevCaseUuid && prevCaseUuid !== newCaseUuid) {
+                            deleteCase(prevCaseUuid).catch((error) =>
+                                handleApiCallError(error)
+                            );
+                        }
+                        setValue(CASE_UUID, newCaseUuid);
+
+                        getCurrentCaseImportParams(newCaseUuid);
+                    })
+                    .catch(() => {
+                        handleApiCallError(errors.apiCallError);
+                    })
+                    .finally(() => {
+                        setCaseFileLoading(false);
+                    });
+            } else {
+                setError(CASE_FILE, {
+                    type: 'caseFileSize',
+                    message: intl
+                        .formatMessage(
+                            {
+                                id: 'uploadFileExceedingLimitSizeErrorMsg',
+                            },
+                            {
+                                maxSize: MAX_FILE_SIZE_IN_MO,
+                                br: <br />,
+                            }
+                        )
+                        .toString(),
+                });
+            }
+        }
+    };
+
+    // handle check studyName
+    const [studyNameAdornment, studyNameChecking] = useNameCheck({
+        field: STUDY_NAME,
+        name: studyName,
+        elementType: ElementType.STUDY,
+        setError,
+    });
+
+    /* Effects */
+    // handle create study from existing case
+    useEffect(() => {
+        if (providedExistingCase) {
+            const { elementUuid } = providedExistingCase;
+            setValue(CASE_FILE, providedExistingCase);
+            setValue(CASE_UUID, elementUuid);
+
+            getCurrentCaseImportParams(elementUuid);
+        }
+    }, [getCurrentCaseImportParams, providedExistingCase, setValue]);
+
+    // handle set study name
+    useEffect(() => {
+        if (caseFile && !apiCallErrorMessage && !caseFileErrorMessage) {
+            const { name: caseFileName } = caseFile;
+
+            if (caseFileName) {
+                clearErrors(STUDY_NAME);
+                setValue(
+                    STUDY_NAME,
+                    caseFileName.substring(0, caseFileName.indexOf('.')),
+                    { shouldDirty: true }
+                );
+            }
+        }
+
+        if (providedExistingCase) {
+            const { elementName: existingCaseName } = providedExistingCase;
+            setValue(STUDY_NAME, existingCaseName, { shouldDirty: true });
+        }
+    }, [
+        caseFile,
+        apiCallErrorMessage,
+        caseFileErrorMessage,
+        providedExistingCase,
+        setValue,
+        clearErrors,
+    ]);
+
+    // handle change possibility to create new study
+    useEffect(() => {
+        setIsCreationAllowed(
```

Comment:
When you use useEffect + useState like this, that means you just need an useMemo

Reply:
yes, right!
thank you